Include/cpython/listobject.h

```diff
@@ -44,7 +44,11 @@ PyList_SET_ITEM(PyObject *op, Py_ssize_t index, PyObject *value) {
     PyListObject *list = _PyList_CAST(op);
     assert(0 <= index);
     assert(index < list->allocated);
+#ifdef Py_GIL_DISABLED
+    _Py_atomic_store_ptr_relaxed(&list->ob_item[index], value);
+#else
     list->ob_item[index] = value;
+#endif
```

Comment:
Can the change to `PyList_SET_ITEM` be reverted now that the problematic call sites use `FT_ATOMIC_STORE_PTR_RELEASE`?